isomerpages/isomercms-frontend
# fix(deps): upgrade design system

## Problem

Closes IS-283

Currently, we are on design system 0.12. However, our frontend still has some issues like 
1. Inconsistent tokens between figma (designer) and engineers 
2. Inability to use [design.hack.gov.sg](http://design.hack.gov.sg/) since there are a lot of breaking changes in version 1.x compared to 0.x. This lead to some visual changes that we experienced as a team (eg. purple background for certain menu components) when we tried to develop components. 
3. We are not using some of design system's best practices. (The primary concern here is that design system does not export project specific tokens, this PR follows the instructions presented [here](https://github.com/opengovsg/design-system/tree/main/token-gen))


## Solution

This issue explores a fix to 

1. upgrade the design system 
2. remove all breaking changes 
3. Add typings for our custom theme tokens 
4. follow best practices that design system was intended to use by following instructions [here](https://github.com/opengovsg/design-system/tree/main/token-gen))


**Breaking Changes**



  - Some of the tokens are no longer being used, therefore they will be modified to fit the newer tokens

**List of changes**
In order to minimise risk, a comprehensive list of changes + checks are given below. 

**Change**:  Remove global css being injected through `bootstrap` 
_Check_: Went to `isomer-templates` to see if we were using `bootstrap` -> global search on `bootstrap` revealed nothing, so this is a safe removal. for cms, our components should be using design system stuff anyway. 

**Change**: Updated broken imports
_Check_: Went to component call site, no ts errors. On cms, made sure they can render. A comprehensive list of updated imports are as follows: 
1. `FormFieldMessage` -> `FormHelperText`
2. `InlineMessage` -> `Infobox` 
3. `GovtMasthead` -> `RestrictedGovtMasthead` 

**Change**: Removal of `useMarkdown` prop, changed to use components instead. 
_Check_: Went to component call site, no ts errors. On cms, made sure they can render. A comprehensive list of removals are as follows: 
1. `LogoSettings` 
2. `ReviewRequestForm`
3. `GeneralSettings`
4. `PageSettingsModal`
5. `ResourcePageSettingsModal`

see [here](https://github.com/opengovsg/design-system/issues/440) for why this happens. We have to do `ts-ignore` otherwise it will be compile error as the type is incompatible. have double checked with @karrui on this point. 

**Change**: updated `useToastProps` to `useToastOptions` 
_Check_: design system no longer exports the former, and now uses the latter under the hood. changed import + updated `danger` to `error`. Toast on error still renders fine. 

**Change**: Updated searchbar to take `onExpansion` as the old prop was removed 
_Check_: Click the search bar + type something, should work

**Change**: Colours are now generated with token -> this was merged with our existing colours
_Check_: Go to cms, it should look like the old one. This was required otherwise our existing colour tokens **will point to invalid reference.** As designers have not finished mapping old -> new tokens, we will use the old ones for now.

**Other misc changes**: 
1. updated imports in stories
2. destructure array for `useToken` 
3. remove auto uppercase for sidebar
4. change `zIndex` for `header` (otherwise, will cut off toast)

**Manual tests**
See above checks section. Additionally, go through **each of the major flows on CMS**: 
1. CRUD file (unlinked, resources, folders, subfolders) 
2. CRUD folders (inclusive of resources, images, files)
3. RR process
4. idk if got anymore pls add if have

## Special notes 
BIG thank you to @karrui for lending his expertises in this PR
